### PR TITLE
#2622-fix

### PR DESCRIPTION
**Pull request by MishaHMK**
closes https://github.com/ita-social-projects/EPlast/issues/2622

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs [hunk 1]
```diff
@@ -78,7 +79,8 @@ private async Task<bool> HasAccessAsync(string userId)
         [HttpGet("degree")]
         public async Task<IActionResult> GetAllDergees()
         {
-            return Ok(await _plastDegreeService.GetDergeesAsync());
+            var degrees = await _plastDegreeService.GetDergeesAsync();
+            return Ok(degrees.OrderBy(d => d.Name));
```
Comment (hryhoriivoloshyn) [inline EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs:83]: It is better to add sorting to service level, not a controller.

**Review by hryhoriivoloshyn — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs (hunk 1 above)
Comment (MishaHMK) [reply EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs:83]: ok, thanks

File: EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs [hunk 2]
```diff
@@ -78,7 +79,8 @@ private async Task<bool> HasAccessAsync(string userId)
         [HttpGet("degree")]
         public async Task<IActionResult> GetAllDergees()
         {
-            return Ok(await _plastDegreeService.GetDergeesAsync());
+            var degrees = await _plastDegreeService.GetDergeesAsync();
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/ActiveMembershipController.cs:82]: в чому сенс

File: EPlast/EPlast.WebApi/Controllers/RegionsController.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/RegionsController.cs:3]: Навіщо додано

File: EPlast/EPlast.WebApi/Controllers/ClubController.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
 using System;
+using System.Linq;
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/ClubController.cs:2]: Навіщо додано

File: EPlast/EPlast.BLL/Services/Region/RegionService.cs [hunk 5]
```diff
@@ -388,7 +388,8 @@ public async Task<IEnumerable<RegionForAdministrationDto>> GetRegions(UkraineObl
         public IEnumerable<RegionNamesDto> GetActiveRegionsNames()
         {
             var regions = _repoWrapper.Region
-                .GetActiveRegionsNames();
+                .GetActiveRegionsNames().OrderBy(r => r.RegionName);
```
Comment (tvardero) [inline EPlast/EPlast.BLL/Services/Region/RegionService.cs:391]: Перенеси сортування до GetActiveRegionsNames

File: EPlast/EPlast.BLL/Services/Club/ClubService.cs [hunk 6]
```diff
@@ -370,12 +370,18 @@ private async Task<bool> CheckCreated(string name)
                 predicate: a => a.Name == name) != null;
         }
 
+        private Func<IQueryable<DataAccessClub.Club>, IQueryable<DataAccessClub.Club>> GetOrder()
```
Comment (hryhoriivoloshyn) [inline EPlast/EPlast.BLL/Services/Club/ClubService.cs:373]: Same here, just write lambda expression, no need in separate functions.

**Review by hryhoriivoloshyn — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Services/ActiveMembership/PlastDegreeService.cs [hunk 7]
```diff
@@ -25,12 +25,19 @@ public PlastDegreeService(IMapper mapper, IRepositoryWrapper repoWrapper, IUserM
             _userManagerService = userManagerService;
         }
 
+        private Func<IQueryable<PlastDegree>, IQueryable<PlastDegree>> GetOrder()
```
[the remark this answers is not in the document]
Comment (tvardero) [reply EPlast/EPlast.BLL/Services/ActiveMembership/PlastDegreeService.cs]: @hryhoriivoloshyn I WILL HAVE ORDER

**Review by hryhoriivoloshyn — APPROVED** [review]

**Review by tvardero — DISMISSED** [review]
Сквашни коммиты

File: EPlast/EPlast.BLL/Services/Club/ClubService.cs [hunk 8]
```diff
@@ -373,9 +373,9 @@ private async Task<bool> CheckCreated(string name)
         /// <inheritdoc />
         public async Task<IEnumerable<ClubForAdministrationDto>> GetClubs()
         {
-            var clubs = await _repoWrapper.Club.GetAllAsync();
-            var filteredClubs = clubs.Where(c => c.IsActive);
-            return _mapper.Map<IEnumerable<DataAccessClub.Club>, IEnumerable<ClubForAdministrationDto>>(filteredClubs);
+            var clubs = await _repoWrapper.Club.GetRangeAsync(c => c.IsActive, null, x => x.OrderBy(e => e.Name), null, null, null);
```
[the remark this answers is not in the document]
Comment (MishaHMK) [reply EPlast/EPlast.BLL/Services/Club/ClubService.cs:376]: Я так зробив, бо з поіменованими параметрами юніт-тести сварилися, що я їм повертаю null звідси

**Review by Serhii-Buniak — APPROVED** [review]

**Review by gutovskii — APPROVED** [review]

**Pull request merged**